src/watchdog/observers/read_directory_changes.py

```diff
@@ -62,8 +63,11 @@ def __init__(self, event_queue, watch, timeout=DEFAULT_EMITTER_TIMEOUT):
         self._lock = threading.Lock()
         self._handle = None
 
-    def on_thread_start(self):
+    def start(self):
         self._handle = get_directory_handle(self.watch.path)
+        super().start()
+        if platform.python_implementation() != 'CPython':
+            time.sleep(0.01)  # Workaround for PyPy https://github.com/gorakhargosh/watchdog/issues/792
```

Comment:
Why not keeping the original code to not break other's code? Also, it may be better to keep a method only for PyPy.
Something like:
```python
    def on_thread_start(self):
        self._handle = get_directory_handle(self.watch.path)

    if platform.python_implementation() == 'PyPy':
        def start(self):
            """PyPy needs some time before receiving events, see #792."""
            super().start()
            time.sleep(0.01)
```